File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/HygieneGameTests.java
package dev.krysztal.casualtiesbelow.gametest;

import net.fabricmc.fabric.api.gametest.v1.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;

/** Java discovery bridge for the Scala hygiene GameTest scenarios. */
public final class HygieneGameTests {
    @GameTest(maxTicks = 20)
    public void dirtinessPassiveAccrualAccumulates(GameTestHelper helper) {
        HygieneScenarios.passiveAccrualAccumulates(helper);
    }

    @GameTest(maxTicks = 20)
    public void cauldronSoakWashesAndConsumesLevels(GameTestHelper helper) {
        HygieneScenarios.cauldronSoakWashesAndConsumesLevels(helper);
    }

    @GameTest(maxTicks = 20)
    public void waterWashRemovesDirtiness(GameTestHelper helper) {
        HygieneScenarios.waterWashRemovesDirtiness(helper);
    }

    @GameTest(maxTicks = 20)
    public void dirtyInjectionSeedsInfection(GameTestHelper helper) {
        HygieneScenarios.dirtyInjectionSeedsInfection(helper);
    }
}

File: src/gametest/scala/dev/krysztal/casualtiesbelow/gametest/HygieneGameTests.scala
package dev.krysztal.casualtiesbelow.gametest

import net.minecraft.gametest.framework.GameTestHelper

import net.fabricmc.fabric.api.gametest.v1.GameTest

/** GameTest discovery entrypoint for the hygiene scenarios ([[HygieneScenarios]]). */
final class HygieneGameTests {

  @GameTest(maxTicks = 20)
  def dirtinessPassiveAccrualAccumulates(helper: GameTestHelper): Unit = {
    HygieneScenarios.passiveAccrualAccumulates(helper)
  }

  @GameTest(maxTicks = 20)
  def dirtAccumulationSettingScalesPassiveGain(helper: GameTestHelper): Unit = {
    HygieneScenarios.dirtAccumulationSettingScalesPassiveGain(helper)
  }

  @GameTest(maxTicks = 20)
  def cauldronSoakWashesAndConsumesLevels(helper: GameTestHelper): Unit = {
    HygieneScenarios.cauldronSoakWashesAndConsumesLevels(helper)
  }

  @GameTest(maxTicks = 20)
  def waterWashRemovesDirtiness(helper: GameTestHelper): Unit = {
    HygieneScenarios.waterWashRemovesDirtiness(helper)
  }

  @GameTest(maxTicks = 20)
  def dirtyInjectionSeedsInfection(helper: GameTestHelper): Unit = {
    HygieneScenarios.dirtyInjectionSeedsInfection(helper)
  }

  @GameTest(maxTicks = 20)
  def disabledInfectionsBlockDirtyNeedleOnset(helper: GameTestHelper): Unit = {
    HygieneScenarios.disabledInfectionsBlockDirtyNeedleOnset(helper)
  }
}
